sklearn/neighbors/_graph.py

```diff
@@ -342,12 +365,6 @@ class KNeighborsTransformer(
     (178, 178)
     """
 
-    _parameter_constraints: dict = {
```

Comment:
for classes, parameter validation for the constructor arguments is done via `_parameter_constraints` class attribute as you see here, and not via the `@validate_params` decorator.

And as you can see, it is implemented already for this class, so nothing left to be done for `KNeighborsTransformer`.

`@validate_params` is used for functions, and we're adding them to most public functions, for example, this one: https://github.com/scikit-learn/scikit-learn/pull/26086. Therefore this PR can be closed.